csharp/src/Ice/SlicBinaryConnection.cs

```diff
@@ -107,164 +161,231 @@ public async ValueTask CloseAsync(Exception exception, CancellationToken cancel)
 
         public async ValueTask<(long StreamId, IncomingFrame? Frame, bool Fin)> ReceiveAsync(CancellationToken cancel)
         {
+            FrameType type;
+            ArraySegment<byte> data;
             while (true)
             {
-                ArraySegment<byte> buffer = await PerformReceiveFrameAsync().ConfigureAwait(false);
-                (int requestId, IncomingFrame? frame) = ParseFrame(buffer);
-                if (frame != null)
+                // Read Slic frame header
+                (type, data) = await ReceiveFrameAsync(CancellationToken.None).ConfigureAwait(false);
+                switch (type)
                 {
-                    return (StreamId: requestId, Frame: frame, Fin: requestId == 0 || frame is IncomingResponseFrame);
+                    case FrameType.Ping:
+                    {
+                        ValueTask task = PrepareAndSendFrameAsync(FrameType.Pong, null, CancellationToken.None);
```

Comment:
The frame type does not look correct.

Reply:
Hmm I don't see why, we send a Pong frame after receiving a Ping frame.